lib/AST/Decl.cpp

```diff
@@ -1527,6 +1527,47 @@ InheritedEntry::InheritedEntry(const TypeLoc &typeLoc)
     isUnchecked = typeRepr->findUncheckedAttrLoc().isValid();
 }
 
+InheritedTypes::InheritedTypes(
+    llvm::PointerUnion<const TypeDecl *, const ExtensionDecl *> decl)
+    : Decl(decl) {
+  if (auto *typeDecl = decl.dyn_cast<const TypeDecl *>()) {
+    Entries = typeDecl->Inherited;
+  } else {
+    Entries = decl.dyn_cast<const ExtensionDecl *>()->Inherited;
```

Comment:
since you unconditionally dereference, I think that can be `cast` (or it's called `get` IIRC)